File: src/typescript/src/schemas.ts
export interface Message {
  id: number;
  agent_id: string;
  content: string;
  processed: boolean;
  role?: string;
  name?: string;
  registered_at?: Date;
  processed_at?: Date;
  metadata?: Record<string, any>;
}

export interface File {
  id: number;
  agent_id: string;
  file_path: string;
  file_hash: string;
  size: number;
  last_modified: Date;
  processed: boolean;
  label: string;
  description: string;
  registered_at?: Date;
  processed_at?: Date;
}

export interface MessageCreate {
  content: string;
  role: string;
  name?: string;
  metadata?: Record<string, any>;
}

File: src/python/schemas.py
from datetime import datetime
from typing import List, Dict, Any, Optional
from pydantic import BaseModel


class Message(BaseModel): 
    """ Represents a message in the conversation """
    id: int
    agent_id: str
    content: str
    processed: bool
    role: Optional[str] = None
    name: Optional[str] = None
    registered_at: Optional[datetime] = None
    processed_at: Optional[datetime] = None
    metadata: Optional[Dict[str, Any]] = None


class File(BaseModel):
    """ Represents a file in the database """
    id: int
    agent_id: str
    file_path: str
    file_hash: str
    size: int
    last_modified: datetime
    processed: bool
    label: str
    description: str
    registered_at: Optional[datetime] = None
    processed_at: Optional[datetime] = None


class MessageCreate(BaseModel):
    """ Data for creating a new message """
    content: str
    role: str
    name: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None

